### ISSUE TITLE
ODL configuration for non-JRF domains

### ISSUE THREAD

**Issue by zeusbaba** [oracle/weblogic-deploy-tooling#1506]
We are using a _custom_ `typedef` to configure base domain. 
However, WDT3 doesnot configure ODL anymore, this used work fine with `v1.9.20`  

I notice this in WDT logs when trying to apply ODL config  
``` 
 <OdlHelper> <configure_odl> <WLSDPLY-19709> <ODL configuration for non-JRF domains is not supported, skipping>
``` 

could you please on how to resolve this?

**Comment by robertpatrick** [comment]
This looks like a bug.  WDT needs to know that ODL is installed and currently, it looks at the specified `-domain_type` argument to make sure it was either `JRF` or `RestrictedJRF` (which is not right since it is missing the `JRF-Compact` domain type).  

Probably a better way to accomplish this is to look at the typedef's `rcuSchemas` filed to make sure the list is not empty.  We will get this fixed and into the 3.3.1 release...

**Comment by zeusbaba** [comment]
thanks for your feedback @robertpatrick 
Looking fwd for a fix.  
fyi. as you _might_ recall from my previous ticket, we are in process of upgrade to `WDT3` across all MW components (for our customer), therefore making things work properly is an important factor.

**Comment by zeusbaba** [comment]
fyi. as a temporary workaround, I did update one of our custom `typedef`s to include `"Oracle Restricted JRF"`  , see snippet below
```  
...
 "definitions": {
    "OAM_12c": {
      "baseTemplate": "Basic WebLogic Server Domain",
      "extensionTemplates": [
        "Oracle Restricted JRF",
        "Oracle Access Management Suite",
        "Oracle JRF WebServices Asynchronous services"
      ],
      "customExtensionTemplates": [ ],
      "rcuSchemas": [
        "IAU", "IAU_APPEND", "IAU_VIEWER",
        "OAM", "MDS", "OPSS",
        "STB", "WLS"
      ],
      "serverGroupsToTarget": [ "JRF-MAN-SVR", "WSM-CACHE-SVR", "JRF-WS-CORE-MAN-SVR"],
      "dynamicClusterServerGroupsToTarget" : [ "WSM-CACHE-DYN-CLUSTER" ]
    }
  }
...
```  

it resolves the issue for now, as you shared, we'll be waiting for new release with a proper fix

**Comment by robertpatrick** [comment]
@zeusbaba I would advise against that.  I am pretty sure that `Oracle Access Management Suite` likely already depends on `Oracle JRF` so putting the Restricted JRFG template may do bad things...

**Comment by zeusbaba** [comment]
as you suggested; using `Oracle JRF`  while waiting for `WDT v3.3.1` 
the current `typedef` 
```
"extensionTemplates": [
        "Oracle JRF",
        "Oracle Access Management Suite",
        "Oracle JRF WebServices Asynchronous services"
      ],
```

**Comment by robertpatrick** [comment]
Fixed in 3.4.0

**Comment by zeusbaba** [comment]
thnx @robertpatrick  
i must say it loud;  your team is doing an exemplary good job.  
fast response, good collaboration, and quick resolutions.  
thanks for your efforts
